components/script/dom/xmlhttprequest.rs

```
@@ -564,13 +564,14 @@ impl XMLHttpRequestMethods<crate::DomTypeHolder> for XMLHttpRequest {
                 };
                 let total_bytes = bytes.len();
                 let global = self.global();
-                let stream = ReadableStream::new_from_bytes(&global, bytes, can_gc);
-                Some(ExtractedBody {
-                    stream,
-                    total_bytes: Some(total_bytes),
-                    content_type: Some(DOMString::from(content_type)),
-                    source: BodySource::Object,
-                })
+                ReadableStream::new_from_bytes(&global, bytes, can_gc)
+                    .ok()
```

Should we return an error here (and below) instead with `?`?